domain/schema/model/sql/0007-charm.sql

```diff
@@ -1,27 +1,398 @@
+    (0, 'sha256');
+
+CREATE TABLE charm_hash (
+    charm_uuid          TEXT NOT NULL,
+    hash_kind_id        TEXT NOT NULL,
+    hash                TEXT NOT NULL,
+    CONSTRAINT          fk_charm_hash_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    CONSTRAINT          fk_charm_hash_kind
+        FOREIGN KEY     (hash_kind_id)
+        REFERENCES      hash_kind(id)
+);
+
+CREATE TABLE charm_relation_kind (
+    id       INT PRIMARY KEY,
+    name     TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_charm_relation_kind_name
+    ON charm_relation_kind (name);
+
+INSERT INTO charm_relation_kind VALUES
+    (0, 'provides'), 
+    (1, 'requires'),
+    (2, 'peers');
+
+CREATE TABLE charm_relation_role (
+    id       INT PRIMARY KEY,
+    name     TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_charm_relation_role_name
+    ON charm_relation_role (name);
+
+INSERT INTO charm_relation_role VALUES
+    (0, 'provider'),
+    (1, 'requirer'),
+    (2, 'peer');
+
+CREATE TABLE charm_relation_scope (
+    id       INT PRIMARY KEY,
+    name     TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_charm_relation_scope_name
+    ON charm_relation_scope (name);
+
+INSERT INTO charm_relation_scope VALUES
+    (0, 'global'),
+    (1, 'container');
+
+CREATE TABLE charm_relation (
+    charm_uuid          TEXT NOT NULL,
+    kind_id             TEXT NOT NULL,
+    name                TEXT,
+    role_id             TEXT,
+    interface           TEXT,
+    optional            BOOLEAN,
+    "limit"             INT,
+    scope_id            TEXT,
+    CONSTRAINT          fk_charm_relation_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    CONSTRAINT          fk_charm_relation_kind
+        FOREIGN KEY     (kind_id)
+        REFERENCES      charm_relation_kind(id),
+    CONSTRAINT          fk_charm_relation_role
+        FOREIGN KEY     (role_id)
+        REFERENCES      charm_relation_role(id),
+    CONSTRAINT          fk_charm_relation_scope
+        FOREIGN KEY     (scope_id)
+        REFERENCES      charm_relation_scope(id),
+    PRIMARY KEY (charm_uuid, kind_id, name)
+);
+
+CREATE INDEX idx_charm_relation_charm
+ON charm_relation (charm_uuid);
+
+CREATE TABLE charm_extra_binding (
+    charm_uuid          TEXT NOT NULL,
+    name                TEXT NOT NULL,
+    CONSTRAINT          fk_charm_extra_binding_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    PRIMARY KEY (charm_uuid, name)
+);
+
+CREATE INDEX idx_charm_extra_binding_charm
+ON charm_extra_binding (charm_uuid);
+
+CREATE TABLE charm_category (
+    charm_uuid          TEXT NOT NULL,
+    value               TEXT NOT NULL,
+    CONSTRAINT          fk_charm_category_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    PRIMARY KEY (charm_uuid, value)
 );
 
+CREATE INDEX idx_charm_category_charm
+ON charm_category (charm_uuid);
+
+CREATE TABLE charm_tag (
```

Comment:
What's the difference between category and tag. "tag" in metadata.yaml is "databases" etc. What's category.